### PR TITLE
Allow non-Webpack Workers Sites

### PR DESCRIPTION
**Pull request by GregBrimble**
Right now, custom build commands don't work with Workers SItes. This should enable them.

### CODE REVIEW

File: src/settings/toml/manifest.rs [hunk 1]
```diff
@@ -327,7 +321,7 @@ impl Manifest {
         Not inherited: Must be defined for every environment individually.
         */
         let mut target = Target {
-            target_type,                                 // Top level
```
Comment (xortive) [inline src/settings/toml/manifest.rs:330]: top level is correct -- we don't let you override target_type in environments

**Review by xortive — COMMENTED** [review]
Could we:
* forbid the rust type (it won't work)
* for the javascript type fail if no `[build]`, or a `[build]` section with no/empty `command`? Something like:

```
Workers Sites requires using a bundler, and your configuration indicates that you aren't using one. You can fix this by:
* setting your project type to "webpack" to use our automatically configured webpack bundler.
* setting your project type to "javascriot", and configuring a build command in the `[build]` section if you wish to use your choice of bundler.
```

**Review by xortive — APPROVED** [review]

File: src/upload/form/mod.rs [hunk 2]
```diff
@@ -89,7 +97,7 @@ pub fn build(
                     log::info!("Plain JavaScript project detected. Publishing...");
                     let package_dir = target.package_dir()?;
                     let package = Package::new(&package_dir)?;
-                    let script_path = package.main(&package_dir)?;
+                    let script_path = package_dir.join(package.main(&package_dir)?);
```
Comment (xortive) [inline src/upload/form/mod.rs:100]: why the change here?

File: src/settings/toml/manifest.rs [hunk 3]
```diff
@@ -312,11 +312,32 @@ impl Manifest {
...
+                TargetType::Rust => {
+                    failure::bail!(format!(
+                        "{} Workers Sites does not support Rust type projects.",
+                        emoji::WARN
+                    ))
+                }
+                TargetType::JavaScript => {
+                    if let Some(build) = &self.build {
+                        if build.command.is_none() {
+                            failure::bail!(format!(
```
Comment (xortive) [inline src/settings/toml/manifest.rs:326]: nit: would be nice if this message wasn't duplicated but I'm not too bothered by it if it's inconvenient to structure the code that way

File: src/upload/form/mod.rs (hunk 2 above)
Comment (GregBrimble) [reply src/upload/form/mod.rs:100]: Without it, `script_path` is relative to the entrypoint. By joining `package_dir` with `script_path`, we get an absolute path, which is needed since wrangler actually executes in the project directory (parent).

So yes, it's needed, otherwise wrangler can't find that script to deploy.

**Review by xortive — APPROVED** [review]

**Review by ytbigvinny27 — COMMENTED** [review]
🔜🔜🔜🔝
